### PR TITLE
adding tests

### PR DESCRIPTION
**Pull request by thejacer87**

### CODE REVIEW

File: tests/CanadaPost/Tests/RatingTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            foreach ($quotes as $rate) {
+                $this->assertTrue(is_array($rate));
+                $this->assertNotEmpty($rate['service-code']);
+                $this->assertNotEmpty($rate['service-name']);
+                $this->assertGreaterThan(0, $rate['price-details']['base']);
+            }
+        }
+    }
+
+}
```
Comment (krystalcode) [inline tests/CanadaPost/Tests/RatingTest.php]: You need to be clear on what exactly you are testing here: that the response was properly parsed and the values are not altered in any way. Checking if `$rate` is an array (you could also check `price-quotes`) and checking if we do have values is ok, it checks that we have the expected structure. You do know however the exact expected values, you provided them, so check that they are equal to ensure that the values are not altered in any way while parsing the response.

Comment (krystalcode) [reply tests/CanadaPost/Tests/RatingTest.php]: That means, for example, `$this->assertEquals('9.59', $rate['price-details']['base']);`

Comment (thejacer87) [reply tests/CanadaPost/Tests/RatingTest.php]: to test all the price values i would have to only test one rate then.

or remove the foreach and test something like this:

`$this->assertEquals('9.59', $rates[0]['price-details']['base']);`
`$this->assertEquals('22.64', $rates[1]['price-details']['base']);`
`$this->assertEquals('9.59', $rates[2]['price-details']['base']);`
`$this->assertEquals('12.26', $rates[2]['price-details']['base']);`

is that what you want?

File: src/ClientBase.php [hunk 2]
```diff
@@ -52,10 +111,8 @@ public function get($endpoint, array $headers = [])
             }
```
Comment (krystalcode) [inline src/ClientBase.php]: I think I previously had a comment about allowing to override the debug option, now that we support passing options:

```
if (!isset($options['debug']) && $this->config['env'] === self::ENV_DEVELOPMENT) {
  $options['debug'] = true;
}
```

Using capitals for `true/false` btw is a Drupal convention, general in PHP libraries we should be using lowercase (that was my code actually).

File: src/ClientBase.php [hunk 3]
```diff
@@ -88,10 +164,8 @@ public function post($endpoint, array $headers = [], $payload)
...
```
Comment (krystalcode) [inline src/ClientBase.php]: Same comment as above.

File: src/ClientBase.php (hunk 2 above)
Comment (krystalcode) [reply src/ClientBase.php]: I see you have added the code to the `buildClient` function, it should be removed from here then.

**Pull request merged**